lib/new_post_manager.rb

```diff
@@ -318,4 +318,18 @@ def perform_create_post
 
     result
   end
+
+  private
+
+  def self.allowed_groups
+    if SiteSetting.approve_unless_allowed_groups.is_a?(String)
```

Comment:
What is the type of `SiteSetting.approve_unless_allowed_groups`? Isn't it always a group list?

Reply:
Good question! One would think it would always be a group list. For a single group it returns an integer. If there are multiple groups it returns a pipe delimited string like "10|11". I was getting errors originally because it was trying to `.split` an Integer.

Reply:
@oblakeerickson no need for all this, `group_list` site settings now have a magical `_map` version of the setting name defined here:

https://github.com/discourse/discourse/blob/4072dcd7d58c4f376c6944efbbc1805eb44205bc/lib/site_setting_extension.rb#L536-L543

It handles empty strings & split. So you can do `SiteSetting.approve_unless_allowed_groups_map`

Reply:
@martin-brennan oh way cool!